utils/resb/src/bundle.rs

```diff
@@ -0,0 +1,326 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! The `bundle` module provides data structures for working directly with the
+//! contents of a resource bundle.
+//!
+//! WARNING: This module is not suitable for use at runtime due to its reliance
+//! on `std` and `alloc` and therefore not intended for general deserialization
+//! of resource bundles. Rather, it is intended to be used in development-time
+//! tools for working with bundles.
+
+extern crate alloc;
+use alloc::{
+    borrow::Cow,
+    collections::{btree_map, BTreeMap},
+};
+
+use crate::MASK_28_BIT;
+
+/// A tree-like collection of data [`Resource`]s primarily intended for storing
+/// locale and other internationalization data for [ICU] (International
+/// Components for Unicode).
+///
+/// [ICU]: https://icu.unicode.org/
+#[derive(Debug)]
+pub struct ResourceBundle<'a> {
+    name: Cow<'a, str>,
+    root: Resource<'a>,
+    is_locale_fallback_enabled: bool,
+}
+
+impl<'a> ResourceBundle<'a> {
+    /// Makes a new resource bundle with the specified resource at its root.
+    pub fn new(name: Cow<'a, str>, root: Resource<'a>, is_locale_fallback_enabled: bool) -> Self {
+        Self {
+            name,
+            root,
+            is_locale_fallback_enabled,
+        }
+    }
+
+    /// Gets the name of the resource bundle.
+    pub fn name(&self) -> &str {
```

Comment:
Why do you install those getters? You can just expose the properties and as long as the type is non-mutable, the values cannot be overwritten. In Rust we don't use getters to limit writeability.

Reply:
Partly addressed in https://github.com/unicode-org/icu4x/pull/4267.